### PR TITLE
enable maxCensusSize and upgrade genesis format

### PR DESCRIPTION
**Pull request by p4u**
Specifying maxCensusSize is a requirement now when creating an election. The size is the number of voters expected or the maximum that an election organizer wants to allow.

There might be a maximum size in the blockchain, defined in the genesis file.

### CODE REVIEW

**Comment by p4u** [comment]
missing:
- [x] write a unit test
- [x] API endpoints

File: vochain/transaction/election_tx.go [hunk 1]
```diff
@@ -133,12 +147,12 @@ func (t *TransactionHandler) NewProcessTxCheck(vtx *vochaintx.VochainTx,
 	}
 
 	if tx.Process.Mode.PreRegister &&
-		(tx.Process.MaxCensusSize == nil || tx.Process.GetMaxCensusSize() <= 0) {
+		(tx.Process.MaxCensusSize == 0 || tx.Process.GetMaxCensusSize() <= 0) {
```
Comment (altergui) [inline vochain/transaction/election_tx.go]: this whole check became redundant, there's no chance at this point of the code that GetMaxCensusSize is <= 0, regardless of Mode.PreRegister

**Comment by altergui** [comment]
i'm reviewing and directly pushing the fixes in this PR as commits to be squashed

**Review by altergui — APPROVED** [review]
LGTM

**Comment by altergui** [comment]
rebased to fix a go.mod conflict

**Pull request merged**